website/docs/r/cdn_frontdoor_endpoint.html.markdown

```diff
@@ -0,0 +1,79 @@
+---
+subcategory: "CDN"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cdn_frontdoor_endpoint"
+description: |-
+  Manages a Frontdoor Endpoint.
+---
+
+# azurerm_cdn_frontdoor_endpoint
+
+Manages a Frontdoor Endpoint.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-cdn-frontdoor"
+  location = "West Europe"
+}
+
+resource "azurerm_cdn_frontdoor_profile" "example" {
+  name                = "example-profile"
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_cdn_frontdoor_endpoint" "example" {
+  name                            = "example-endpoint"
+  cdn_frontdoor_profile_id        = azurerm_cdn_frontdoor_profile.example.id
+  enabled                         = true
+  origin_response_timeout_seconds = 120
+
+  tags = {
+    ENV = "example"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Frontdoor Endpoint. Changing this forces a new Frontdoor Endpoint to be created.
```

Comment:
In testing this it appears that name needs to be globally unique?! Not sure why, though, since it is a child of the AFD profile.

But if that's indeed the case, we should probably add a note on that here

Reply:
I do mention this fact, but in the examples:

**IMPORTANT:** CDN Frontdoor is a **GLOBAL** resource. So make sure that your `${var.prefix}` value is sufficiently random else your resources may have naming collisions and cause unexpected errors when running this example and cause it to fail when applied(e.g. `foo`, `bar` or `example` would not be a good choice for the `${var.prefix}` argument as those values would have a high probability of collision).

It may also be warranted to also add this to the documentation as well.

Reply:
yeah it is clear (or at least was clear to me) that the Front Door profile name (which is the top level Azure resource) needs to get a unique name. It was not clear to me, though, that this also applies to the endpoint name. That's why I thought it might make sense to point this out